packages/react-router/lib/rsc/browser.tsx

```diff
@@ -294,10 +294,10 @@ function createRouterFromPayload({
           hasHydrateFallback: match.hydrateFallbackElement != null,
         };
       },
-      payload.location,
-      undefined,
-      false,
-    ),
+      location: payload.location,
+      basename: payload.basename,
```

Comment:
It wasn't obvious that the basename wasn't being passed through because of the positional args, so I've refactored the `getHydrationData` function to use an options object rather than positional args.